pandas/_libs/tslibs/conversion.pyx

```diff
@@ -307,11 +311,15 @@ cdef maybe_localize_tso(_TSObject obj, tzinfo tz, NPY_DATETIMEUNIT reso):
     if obj.value != NPY_NAT:
         # check_overflows needs to run after _localize_tso
         check_dts_bounds(&obj.dts, reso)
-        check_overflows(obj)
+        check_overflows(obj, reso)
 
 
-cdef _TSObject convert_datetime_to_tsobject(datetime ts, tzinfo tz,
-                                            int32_t nanos=0):
+cdef _TSObject convert_datetime_to_tsobject(
+    datetime ts,
+    tzinfo tz,
+    int32_t nanos=0,
```

Comment:
Just confirming, even if `reso=NPY_FR_s` and `nanos>0`, we still want to apply the supplemental `nanos` to the underlying value?

Reply:
yes, bc we're adding it to obj.dts.ps